spec/asyncapi.md

```diff
@@ -685,12 +735,49 @@ bindings:
       exclusive: true
 tags:
   - name: user
-    description: User-related messages
+    description: User-related channels
 externalDocs:
   description: 'Find more info here'
   url: 'https://example.com'
+traits:
+  - $ref: '#/components/channelTraits/kafka'
+```
+
+
+#### <a name="channelTraitObject"></a>Channel Trait Object
+
+Describes a trait that MAY be applied to an [Channel Object](#channelObject). This object MAY contain any property from the [Channel Object](#channelObject), except the `messages` and `traits` ones.
```

Comment:
Why not `messages`? What if I want to define another channel with the same messages but a different address?

Reply:
This is the same problem as with `payload` in the message. You can't define payload in message traits, and likewise you shouldn't be able to add messages to the channel trait - look at my example in the PR description. The current main problem in defining channels is that we define messages on channel level (this is not a problem at all) but when we want to add a "filter" on the operation to inform user on which messages given operation operates, we need to define another channel with subset of messages - maybe this will fix that problem in more elegant way https://github.com/asyncapi/spec/pull/870

> Why not messages? What if I want to define another channel with the same messages but a different address?

This is some use case, but when we give the ability to define the messages at the level of the channel trait then someone can do something like this:

```yaml
channels:
  someChannel:
    address: ...
    messages:
      someMessage: ...
    traits:
      - messages:
           someMessage: ...
```

The ability to override information on the message from the level of the channel trait is a bad idea, that's why I want to avoid it.